src/common/snippets/src/op/subgraph.cpp

```diff
@@ -31,49 +31,64 @@
 
 using namespace std;
 using namespace ngraph;
+using namespace ov::op::util;
+
+BWDCMP_RTTI_DEFINITION(snippets::op::Subgraph);
 
 void snippets::op::Subgraph::set_generator(std::shared_ptr<ngraph::snippets::Generator> generator) {
     m_generator = generator;
 }
 
 snippets::op::Subgraph::Subgraph(const OutputVector& args, std::shared_ptr<ov::Model> body)
-    : Op(args), m_body(body), m_generator(nullptr) {
+    : SubGraphOp(args) {
+    set_function(body);
     constructor_validate_and_infer_types();
+    auto body_params = body->get_parameters();
+    for (int i = 0; i < body_params.size(); ++i) {
+        m_input_descriptions[0].push_back(std::make_shared<ngraph::op::TensorIterator::InvariantInputDescription>(i, i));
+    }
+    auto outputs = body->get_output_size();
+    for (int i = 0; i < outputs; ++i) {
+        m_output_descriptions[0].push_back(std::make_shared<BodyOutputDescription>(i, i));
+    }
 }
 
 snippets::op::Subgraph::Subgraph(const NodeVector& args, std::shared_ptr<ov::Model> body)
     : Subgraph(as_output_vector(args), body) {}
 
 std::shared_ptr<Node> snippets::op::Subgraph::clone_with_new_inputs(const OutputVector& inputs) const {
     INTERNAL_OP_SCOPE(Subgraph);
-    return make_shared<Subgraph>(inputs, ov::clone_model(*m_body.get()));
+    return make_shared<Subgraph>(inputs, ov::clone_model(body()));
 }
 
 void snippets::op::Subgraph::validate_and_infer_types() {
     INTERNAL_OP_SCOPE(Subgraph);
     OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::validate_and_infer_types")
     ngraph::ParameterVector old_parameters;
-    for (auto op : m_body->get_parameters()) {
+    for (auto op : body().get_parameters()) {
```

Comment:
It looks like `body()` returns a copy of `*m_bodies[0]` on every call. If so, why don't we use `body_ptr()` instead?